src/compose/update-strategies.ts

```diff
@@ -12,12 +13,30 @@ export interface StrategyContext {
 	dependenciesMetForStart: boolean;
 	dependenciesMetForKill: boolean;
 	needsSpecialKill: boolean;
+	currentServices: Service[];
+	currentServicesLocked: boolean;
+	appsToLock: AppsToLockMap;
+}
+
+function generateLockThenKillStep(
```

Comment:
This function is strange, as it either returns a step or updates an array and returns `null`, that is probably a sign that those things should happen in different places

Reply:
The alternative to this is to repeat the contents of `generateLockThenKillSteps` 3 times, once under `download-then-kill`, once under `kill-then-download`, and once under `hand-over`. I didn't want the repetition, and the functionality here is necessary: if we're not generating a kill step, we're modifying the appsToLock map to be parsed after service step generation into a takeLock step.